cachi2/core/package_managers/gomod.py

```diff
@@ -734,19 +734,18 @@ def _validate_local_replacements(modules: List[GoModule], app_path: RootedPath)
         if not Path(module.replace.dir).is_relative_to(app_path.root):
             raise Exception(f"{module.replace.dir} is not relative to {app_path.path}")
 
+        # Hack to set the replacement version
+        # Doing it properly will require a bigger refactoring
+        module.replace.path = _get_golang_version(module.replace.path, app_path)
```

Comment:
I think this line does something like:
```python
before = GoModule(
    path="github.com/foo/locally-replaced-module",
    version="v0.0.0",
    replace=GoModule(
        path="./path/to/locally-replaced-module"
        version=None,
    ),
)
after = GoModule(
    path="github.com/foo/locally-replaced-module",
    version="v0.0.0",
    replace=GoModule(
        path="<main module version>"
        version=None,
    ),
)
```
which doesn't seem right:
* passes the wrong thing to `_get_golang_version(module_name, `
* gets the version of the main module instead of getting the version of the locally replaced module
* sets the version as path rather than version (I guess this is the to-be-refactored part?)

We should probably also pass `update_tags=False` to make sure tags are only updated once